src/main/java/org/zalando/nakadi/domain/StrictJsonParser.java

```diff
@@ -0,0 +1,283 @@
+package org.zalando.nakadi.domain;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class StrictJsonParser {
+
+    private static final String POSSIBLE_NUMBER_DIGITS = "0123456789-.Ee";
+
+    private static class StringTokenizer {
+
+        private int currentPosition;
+        private final String value;
+        private final int endIndex;
+
+        StringTokenizer(final String value, final int from, final int to) {
+            this.value = value;
+            this.currentPosition = from;
+            this.endIndex = to;
+        }
+
+        boolean hasNext() {
+            return currentPosition < endIndex;
+        }
+
+        char next() {
+            if (currentPosition >= endIndex) {
+                throw new JSONException("Unexpected end of data at pos " + currentPosition);
+            }
+            return value.charAt(currentPosition++);
+        }
+
+        int getCurrentPosition() {
+            return currentPosition;
+        }
+
+        String next(final int count) {
+            if (currentPosition + count >= endIndex) {
+                throw new JSONException("Unexpected end of data at pos " + currentPosition);
+            }
+            currentPosition += count;
+            return value.substring(currentPosition - count, currentPosition);
+        }
+
+        char nextUnskippable() {
+            for (; ; ) {
+                final char value = next();
+                if (isEmptyCharacter(value)) {
+                    continue;
+                }
+                return value;
+            }
+        }
+
+        void back() {
+            this.currentPosition--;
+        }
+
+        static boolean isEmptyCharacter(final char c) {
+            return (c == ' ' || c == '\t' || c == '\n' || c == '\r');
+        }
+    }
+
+    public static JSONObject parseObject(final String value) throws JSONException {
```

Comment:
Is it possible to provide more context when showing error to user. Currently it looks like:
```
{
  "type": "http://httpstatus.es/400",
  "title": "Bad Request",
  "status": 400,
  "detail": "Expected false value at pos 164"
}
```
I think it would be better if we show something like "Error occurred when parsing event(s). Expected false value at pos 164"